### ISSUE TITLE
image rescaling and featurewise normalization not coordinated 

### ISSUE THREAD

**Issue by alexhernandezgarcia** [keras-team/keras-preprocessing#100]
I believe the current implementation provides an erroneous, unexpected behaviour if both the `rescale` parameter is used (`not None` or different than 0) and feature-wise normalization is applied (`featurewise_center`, `featurewise_std_normalization`, ZCA whitening). 

 The `fit()` function computes the statistics from the original, _un-rescaled_ inputs and these statistics are applied finally on the _rescaled_ data. For instance, if the images are `uint8` (in the range [0, 255]) the feature-wise mean may be, for instance, 128. Then, if `rescale=1./255` the output images will be in the range [0, 1], but the original mean 128 will be subtracted.

```
import numpy as np
from keras.preprocessing.image import ImageDataGenerator

images = np.random.randint(low=0, high=255, size=(10, 32, 32, 3))

rescale = 1. / 255
imagedatagen = ImageDataGenerator(rescale=rescale,
                                  featurewise_center=True)
imagedatagen.fit(images)
batchgen = imagedatagen.flow(images, batch_size=10)
batch = batchgen.next()

images = images.astype(float)
images *= rescale
mean = np.mean(images, axis=(0, 1, 2))
images -= mean
print('Data range should be (approximately): [{}, {}]. \n'
      'Actual data range is: [{}, {}]'.format(np.min(images),
                                              np.max(images),
                                              np.min(batch),
                                              np.max(batch)))
```
